### PR TITLE
Unittest to pytest

### PR DESCRIPTION
**Pull request by oysteoh**

### CODE REVIEW

File: python/CMakeLists.txt [hunk 1]
```diff
@@ -146,4 +145,4 @@ add_test(NAME python.example.makefile     COMMAND ${python} make-file.py test-da
 add_test(NAME python.example.makepsfile   COMMAND ${python} make-ps-file.py test-data/small-prestack.sgy 10 1 5 1 4 1 3)
 add_test(NAME python.example.subcube      COMMAND ${python} copy-sub-cube.py test-data/small.sgy test-data/copy.sgy)
 add_test(NAME python.example.rotate       COMMAND ${python} make-rotated-copies.py test-data/small.sgy ex-rotate.sgy)
-add_test(NAME python.example.scan_min_max COMMAND ${python} scan_min_max.py test-data/small.sgy)
+add_test(NAME python.example.scan_min_max COMMAND ${python} scan_min_max.py test-data/small.sgy)
```
Comment (jokva) [inline python/CMakeLists.txt]: Fix missing newline

File: python/test/test_segy.py [hunk 2]
```diff
@@ -0,0 +1,1097 @@
+from __future__ import absolute_import
+try:
+    from future_builtins import zip, map
+except ImportError: pass
+
+from types import GeneratorType
+
+import os
```
Comment (jokva) [inline python/test/test_segy.py]: Sort includes

File: python/test/test_segy.py [hunk 3]
```diff
@@ -0,0 +1,1097 @@
...
+from segyio._field import Field
+from segyio._line import Line
+from segyio._header import Header
+from segyio._trace import Trace
+from shutil import copy
+
+
+filename = "test-data/small.sgy"
+prestack = "test-data/small-ps.sgy"
+fileMx1  = "test-data/Mx1.sgy"
```
Comment (jokva) [inline python/test/test_segy.py]: Drop these

File: python/test/test_segy.py [hunk 4]
```diff
@@ -0,0 +1,1097 @@
...
+        assert 4.20024, approx(data[0, sample_count//2-1], rel=1e-6)
+        # last sample
+        assert 4.20049, approx(data[0, -1], rel=1e-6)
+
+        # middle xline
+        middle_line = 2
+        # first sample
+        assert 4.22, approx(data[middle_line, 0], rel=1e-5)
+        # middle sample
+        assert 4.22024, approx(data[middle_line, sample_count//2-1], rel=1e-6)
```
Comment (jokva) [inline python/test/test_segy.py]: abs, not rel (or both)

File: python/test/test_segy.py [hunk 5]
```diff
@@ -0,0 +1,1097 @@
...
+        f.bin.update( ((segyio.su.reno, 3), (segyio.su.hdt, 7)) )
+
+        f.flush()
+        assert 12 == f.bin[segyio.su.jobid]
+        assert  4 == f.bin[segyio.su.lino]
+        assert  3 == f.bin[segyio.su.reno]
+        assert  7 == f.bin[segyio.su.hdt]
+
+        # looking up multiple values at once returns a { TraceField: value } dict
+        assert d ==f.bin[BinField.Traces, BinField.SweepFrequencyStart]
```
Comment (jokva) [inline python/test/test_segy.py]: Space on each side of ==

File: python/test/test_segy.py [hunk 6]
```diff
@@ -0,0 +1,1097 @@
...
+        assert 1 == f.header[0][TraceField.offset]
+        assert 2 == f.header[1][TraceField.offset]
+
+        for x, y in zip(f.iline[:,:], cube):
+            assert list(x.flatten()) == list(y.flatten())
+
+
+@pytest.mark.datafiles(filename)
+def test_create_write_lines(datafiles):
+    fname = "mklines.sgy"
```
Comment (jokva) [inline python/test/test_segy.py]: Unnecessary fname var

File: python/test/test_segyio_c.py [hunk 7]
```diff
@@ -0,0 +1,537 @@
...
+
+    textheader = f.gettext(0)
+    textheader = textheader.decode('ascii')  # Because in Python 3.5 bytes are not comparable to strings
+    assert textheader == "yolo" * 800
+
+    f.close()
+
+
+def get_instance_segyiofd(datafiles, file_name="small.sgy", mode="r+", binary=None):
+    path = str(datafiles)
```
Comment (jokva) [inline python/test/test_segyio_c.py]: remove `str` etc

File: python/test/test_segyio_c.py [hunk 8]
```diff
@@ -0,0 +1,537 @@
...
+
+    trace_header = f.getth(0, mkempty())
+
+    assert _segyio.getfield(trace_header, ilb) == 99
+    assert _segyio.getfield(trace_header, xlb) == 42
+
+    f.close()
+
+
+@pytest.mark.datafiles(small_sgy)
```
Comment (jokva) [inline python/test/test_segyio_c.py]: Don't need datafiles, use `pytest.tmpdir` or similar

File: python/test/test_segyio_enum.py [hunk 9]
```diff
@@ -0,0 +1,21 @@
+from segyio import Enum
```
Comment (jokva) [inline python/test/test_segyio_enum.py]: Kill file

File: python/test/test_tools.py [hunk 10]
```diff
@@ -0,0 +1,217 @@
...
+    lines = {i + 1: chr(64 + i) * 76 for i in range(40)}
+    text_header = segyio.create_text_header(lines)
+
+    for line_no in range(0, 40):
+        line = text_header[line_no * 80: (line_no + 1) * 80]
+        assert line == "C{0:>2} {1:76}".format(line_no + 1, chr(64 + line_no) * 76)
+
+
+def test_native():
+    with open(SMALL_SGY, 'rb') as f, segyio.open(SMALL_SGY) as sgy:
```
[the remark this answers is not in the document]
Comment (oysteoh) [reply python/test/test_tools.py]: It's actually not working without 'rb' - at least tests not running as usually

**Pull request merged**